rust/chains/tw_pactus/src/transaction.rs

```diff
@@ -0,0 +1,410 @@
+        PayloadType::try_from(u8::decode(r)?)
+    }
+}
+
+pub trait Payload: Debug + Encodable {
+    fn signer(&self) -> &Address;
+    fn value(&self) -> Amount;
+    fn payload_type(&self) -> PayloadType;
+}
+
+#[derive(Debug)]
+pub struct TransferPayload {
+    sender: Address,
+    receiver: Address,
+    amount: Amount,
+}
+
+impl Encodable for TransferPayload {
+    fn encode(&self, w: &mut dyn std::io::Write) -> Result<(), EncoderError> {
+        self.sender.encode(w)?;
+        self.receiver.encode(w)?;
+        self.amount.encode(w)?;
+
+        Ok(())
+    }
+
+    fn encoded_size(&self) -> usize {
+        self.sender.encoded_size() + self.receiver.encoded_size() + self.amount.encoded_size()
+    }
+}
+
+impl Decodable for TransferPayload {
+    fn decode(r: &mut dyn std::io::Read) -> Result<Self, EncoderError> {
+        let sender = Address::decode(r)?;
+        let receiver = Address::decode(r)?;
+        let amount = Amount::decode(r)?;
+
+        Ok(TransferPayload {
+            sender,
+            receiver,
+            amount,
+        })
+    }
+}
+
+impl Payload for TransferPayload {
+    fn signer(&self) -> &Address {
+        &self.sender
+    }
+    fn value(&self) -> Amount {
+        self.amount.clone()
+    }
+    fn payload_type(&self) -> PayloadType {
+        PayloadType::Transfer
+    }
+}
+
+#[derive(Debug)]
+pub struct Transaction {
+    flags: u8,
+    version: u8,
+    lock_time: u32,
+    fee: Amount,
+    memo: String,
+    payload: Box<dyn Payload>,
+    signature: Option<Signature>,
+    public_key: Option<PublicKey>,
+}
+
+impl Transaction {
+    pub fn new(
+        flags: u8,
+        version: u8,
+        lock_time: u32,
+        fee: Amount,
+        memo: String,
+        payload: Box<dyn Payload>,
+    ) -> Self {
+        Transaction {
+            flags,
+            version,
+            lock_time,
+            fee,
+            memo,
+            payload,
+            public_key: None,
+            signature: None,
+        }
+    }
+
+    pub fn from_proto(input: &Pactus::Proto::SigningInput) -> SigningResult<Self> {
+        match &input.transaction {
+            None => SigningError::err(SigningErrorType::Error_invalid_params),
+            Some(trx) => {
+                let payload = match &trx.payload {
+                    Pactus::Proto::mod_TransactionMessage::OneOfpayload::transfer(pld) => {
+                        let sender = Address::from_str(&pld.sender)?;
+                        let receiver = Address::from_str(&pld.receiver)?;
+
+                        Box::new(TransferPayload {
+                            sender,
+                            receiver,
+                            amount: Amount(pld.amount),
+                        })
+                    },
+                    Pactus::Proto::mod_TransactionMessage::OneOfpayload::bond(_pld) => {
+                        return SigningError::err(SigningErrorType::Error_not_supported)
+                    },
```

Comment:
Could you please remove `Bond` transaction payload if it's not supported yet? This is to avoid confusion for WalletCore users

Reply:
I fully understand your concern, but I would like to request that we keep this code. Implementing all the functionality in a single PR by one author might not be ideal. I hope that soon another team member will update the code and add the bond transaction. Until then, I hope we can retain this code.